test/E2ETests/E2ETests/AspNetCore/CancellationEndToEndTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Tests;
+using Microsoft.Azure.Functions.Tests.E2ETests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.Functions.Worker.E2ETests.AspNetCore
+{
+    public class CancellationEndToEndTests : IClassFixture<CancellationEndToEndTests.TestFixture>
+    {
+        private readonly TestFixture _fixture;
+
+        public CancellationEndToEndTests(TestFixture fixture, ITestOutputHelper testOutputHelper)
+        {
+            _fixture = fixture;
+            _fixture.TestLogs.UseTestLogger(testOutputHelper);
+        }
+
+        [IgnoreOnNetFxTestRunTheory]
+        [InlineData("HttpWithCancellationTokenNotUsed", "Work completed.", "Succeeded")]
+        [InlineData("HttpWithCancellationTokenIgnored", "TaskCanceledException: A task was canceled", "Failed")]
+        [InlineData("HttpWithCancellationTokenHandled", "Request was cancelled.", "Succeeded")]
+        public async Task HttpTriggerFunctions_WithCancellationToken_BehaveAsExpected(string functionName, string expectedMessage, string invocationResult)
```

Comment:
Were you able to run this with your changes backed out? If so, which scenario behaved differently?

Reply:
I have not, but I think it's a good idea to try that out!